File: server/php/src/Application/Domain/Payments/PaymentDetails/PaymentProduct5001SpecificOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class PaymentProduct5001SpecificOutput
{
    public ?string $accountNumber = null;
    public ?string $authorisationCode = null;
    public ?string $liability = null;
    public ?string $mobilePhoneNumber = null;
    public ?string $operationCode = null;

    public function __construct(
        ?string $liability = null,
        ?string $accountNumber = null,
        ?string $authorisationCode = null,
        ?string $operationCode = null,
        ?string $mobilePhoneNumber = null
    ) {
        $this->liability = $liability;
        $this->accountNumber = $accountNumber;
        $this->authorisationCode = $authorisationCode;
        $this->operationCode = $operationCode;
        $this->mobilePhoneNumber = $mobilePhoneNumber;
    }

    public function toArray(): array
    {
        return [
            'accountNumber' => $this->accountNumber,
            'authorisationCode' => $this->authorisationCode,
            'liability' => $this->liability,
            'mobilePhoneNumber' => $this->mobilePhoneNumber,
            'operationCode' => $this->operationCode,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/payment_product5001_specific_output.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class PaymentProduct5001SpecificOutput
          attr_accessor :account_number, :authorisation_code, :liability,
                        :mobile_phone_number, :operation_code

          def initialize(liability: nil, account_number: nil, authorisation_code: nil,
                         operation_code: nil, mobile_phone_number: nil)
            @liability = liability
            @account_number = account_number
            @authorisation_code = authorisation_code
            @operation_code = operation_code
            @mobile_phone_number = mobile_phone_number
          end

          def to_h
            {
              'accountNumber' => @account_number,
              'authorisationCode' => @authorisation_code,
              'liability' => @liability,
              'mobilePhoneNumber' => @mobile_phone_number,
              'operationCode' => @operation_code
            }
          end
        end
      end
    end
  end
end